File: Uva/UVa10101.py
def bangla_number(x):
    if x == 0:
        return 
    if x // 10000000:
        bangla_number(x // 10000000)
        print(' kuti',end='')
        x %= 10000000
    if x // 100000:
        bangla_number(x /100000)
        print(' lakh',end='')
        x %= 100000
    if x // 1000:
        bangla_number(x / 1000)
        print(' hajar', end='')
        x %= 1000
    if x // 100:
        bangla_number(x // 100 )
        print(' shata', end='')
        x %= 100
    if x:
        print(' {}'.format(x) ,end='')

#main
try:
    label = 1
    while True:
        n = int(input())
        print('{:4d}.'.format(label), end='')
        if n :
            bangla_number(n)
        else:
            print(' 0')
        print()
        label += 1
except EOFError:
    pass

File: UVa_cpp/UVa10101.cpp
#include <iostream>
#include <iomanip>

using namespace std;

void bangla(unsigned long long n){
    // cout << " n = " << n << '\n';
    if (n >= 10000000){
        bangla( n / 10000000);
        n %= 10000000;
        cout << " kuti";
    }
    if (n >= 100000){
        cout << " " << n / 100000 << " lakh";
        return bangla( n % 100000);
    }
    else if (n >= 1000){
        cout << " " << n / 1000 << " hajar";
        return bangla( n % 1000);
    }
    else if (n >= 100){
        cout << " " << n / 100 << " shata";
        return bangla( n % 100);
    }
    else{
        if (n != 0)
            cout << " " << n;
    } 
}


int main(){
    unsigned long long n;
    int count = 0;
    while(cin >> n){
        cout << setw(4) <<++count << ".";
        if ( n == 0 ){
            cout << " 0\n";
            continue;
        } 
        bangla(n);
        cout << '\n';
    }

}